File: glowxq-api/business/business-oj/src/main/java/com/glowxq/oj/meta/enums/ProgramLanguage.java
package com.glowxq.oj.meta.enums;

import com.glowxq.core.common.enums.base.BaseType;
import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * @author glowxq
 * @version 1.0
 * @date 2025/1/20
 */
@AllArgsConstructor
@Getter
public enum ProgramLanguage implements BaseType {

    Cpp("C++", "C++代码"),
    ;

    /**
     * code
     */
    private final String code;

    /**
     * 最近类型
     */
    private final String name;

    /**
     * 根据code获取枚举
     *
     * @param code
     * @return
     */
    public static ProgramLanguage matchCode(String code) {
        for (ProgramLanguage pushStatus : ProgramLanguage.values()) {
            if (pushStatus.getCode().equals(code)) {
                return pushStatus;
            }
        }
        return null;
    }
}

File: oj-web/src/enums/oj/meta/ProgramLanguage.ts
import type { BaseEnum } from '@/enums/base';

// 编程语言枚举接口
interface ProgramLanguageEnum extends BaseEnum {
}

/**
 * 编程语言枚举
 * 用于标识不同的编程语言
 */
export const ProgramLanguage: { [key: string]: ProgramLanguageEnum } = {
  /**
   * C++
   */
  CPP: {
    code: 'C++',
    name: 'C++代码'
  }
};

/**
 * 根据code获取枚举
 * @param code
 * @returns 枚举值或null
 */
export function matchCode(code: string): ProgramLanguageEnum | null {
  return Object.values(ProgramLanguage).find(item => item.code === code) || null;
} 